paasta_tools/cli/cmds/local_run.py

```diff
@@ -687,22 +687,14 @@ def configure_and_run_docker_container(
     )
 
 
-def local_makefile_present():
-    if makefile_responds_to('cook-image'):
-        sys.stderr.write("Local Makefile with 'cook-image' target detected. Assuming --build\n")
-        return True
-    else:
-        return False
-
-
 def paasta_local_run(args):
-    if (args.pull is None) and (args.build is None):
-        # Default to --build if neither is specified
-        if not local_makefile_present():
-            sys.stderr.write("A local Makefile with a 'cook-image' target is required for --build\n")
-            sys.stderr.write("If you meant to pull the docker image from the registry, explicitly pass --pull\n")
-            return 1
```

Comment:
I don't really understand the point of this special case. It seems better to just default to `--build` and let it fail below rather than add more tricky behavior.

Happy to revert this if you disagree.